ipaserver/plugins/permission.py

```diff
@@ -553,13 +554,23 @@ def get_effective_attrs(self, entry):
     def make_aci(self, entry):
         """Make an ACI string from the given permission entry"""
 
+        msg = None
         aci_parts = []
         name = entry.single_value['cn']
 
         # targetattr
         attrs = self.get_effective_attrs(entry)
         if attrs:
             aci_parts.append("(targetattr = \"%s\")" % ' || '.join(attrs))
+        else:
+            # If we are re-parsing an entry for validation then
+            # the effective attrs may be empty.
+            if entry.get('attrs') is None:
+                rights = set(['read', 'write', 'search', 'compare'])
+                if set(entry['ipapermright']).issubset(rights):
```

Comment:
If the permission set is delete,compare the message won't be printed, but compare won't apply to any attr. Is it the intent?  IIRC, add or delete imply write, but I'm not sure for the other permissions.

Reply:
add and delete are entry-level rights (e.g. you can add or delete an entire entry). I'll filter those out before comparing.